cpp/ql/src/semmle/code/cpp/ir/internal/TIRVariable.qll

```diff
@@ -0,0 +1,23 @@
+private import cpp
+private import semmle.code.cpp.ir.implementation.TempVariableTag
+private import semmle.code.cpp.ir.implementation.raw.internal.IRConstruction as Construction
+
+newtype TIRVariable =
```

Comment:
Changing from `FunctionIR` to `Function` has the subtle side effect of creating `IRVariable`s for functions that do not have IR. Mostly, this would be parameters of functions without definitions, and all variable accesses in a non-fully-instantiated template. Since you're already importing `IRConstruction`, it should be easy to restrict these to only functions with IR.

Reply:
Good catch. I've pushed a fix.